src/scipp/optimize/__init__.py

```diff
@@ -94,10 +114,10 @@ def curve_fit(
       >>> da = sc.DataArray(y, coords={'x': x})
 
       >>> from scipp.optimize import curve_fit
-      >>> popt, _ = curve_fit(func, da, p0 = [1.0, 1.0 / sc.Unit('m')])
-      >>> round(popt[0])
+      >>> popt, _ = curve_fit(func, da, p0 = {'b': 1.0 / sc.Unit('m')})
+      >>> round(popt['a'])
       5
-      >>> sc.round(popt[1])
+      >>> sc.round(popt['b'])
       <scipp.Variable> ()    float64            [1/m]  [17]
     """
```

Comment:
The docs don't explain how default values for args are handled. Can you please add a description or an example? (Possibly using `partial`)

Reply:
See update.